dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/mapping/MapController.java

```diff
@@ -155,7 +155,7 @@ public WebMessage putJsonObject( @PathVariable String uid, @CurrentUser User cur
         Map newMap = deserializeJsonEntity( request );
         newMap.setUid( uid );
 
-        mergeService.merge( new MergeParams<>( newMap, map )
+        mergeService.merge( new MergeParams<>( map, newMap )
```

Comment:
how did the "wrong" order manifest itself? was that a bug? 😅

Reply:
Yes it was a bug.
First parameter is used as a base and then all not null fields of the second object will be applied to the first one.
Using the wrong order was resulting in an object with some required fields as null, so persistence was failing.